### PR TITLE
Rebase on top of 2-4-stable

### PR DESCRIPTION
**Pull request by mbj**
Not intended for merge. Just status tracking.

### CODE REVIEW

File: api/app/controllers/spree/api/base_controller.rb [hunk 1]
```diff
@@ -90,10 +89,11 @@ def error_during_processing(exception)
...
         Rails.logger.error exception.backtrace.join("\n")
 
-        error_notifier.call(exception, self) if error_notifier
+        unprocessable_entity(exception.message)
+      end
 
-        render text: { exception: exception.message }.to_json,
-          status: 422 and return
+      def unprocessable_entity(message)
+        render text: { exception: message }.to_json, status: 422
```
Comment (dkubb) [inline api/app/controllers/spree/api/base_controller.rb]: I wonder if we should change this to `status: :unprocessable_entity`

File: api/app/helpers/spree/api/api_helpers.rb [hunk 2]
```diff
@@ -30,7 +30,7 @@ module ApiHelpers
         :stock_item_attributes,
         :promotion_attributes,
         :store_attributes
-      ]
+      ].freeze
```
Comment (dkubb) [inline api/app/helpers/spree/api/api_helpers.rb]: I wonder if we should use `IceNine.deep_freeze` here now.

Comment (mbj) [inline api/app/helpers/spree/api/api_helpers.rb]: The moment we did these freeze calls we where not introducing our dependencies, as we later decided to pull them, I think we can easily backport our deps.

I'll do so the next upstreaming cycle.

**Comment by mbj** [comment]
Closing this for: https://github.com/MountainRoseHerbs/spree/pull/80

**Pull request closed without merging**